server/store/datastore/migration/migration.go

```diff
@@ -128,7 +128,11 @@ func runTasks(sess *xorm.Session, tasks []task) error {
 	return nil
 }
 
-func syncAll(sess *xorm.Session) error {
+type syncEngine interface {
+	Sync2(beans ...interface{}) error
+}
+
+func syncAll(sess syncEngine) error {
```

Comment:
Why do we need this interface?

Reply:
To pass engine and session without having same func wtice

Reply:
Tld: postgrss do not allow us to use sync2 twice in one transaction at all.
mysql does have problems with that too, but tjere i worked around :)